android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -103,125 +66,19 @@ public void init() {
         }
     }
 
-    public void getDefaultAccountPerCoin(
-            Callbacks.Callback1<Set<AccountInfo>> defaultAccountPerCoins) {
-        synchronized (mLock) {
-            if (mKeyringService == null || mBraveWalletService == null) {
-                return;
-            }
-            List<Integer> coins = new ArrayList<>();
-            for (CryptoAccountTypeInfo cryptoAccountTypeInfo :
-                    mSharedData.getSupportedCryptoAccountTypes()) {
-                coins.add(cryptoAccountTypeInfo.getCoinType());
-            }
-            mKeyringService.getAllAccounts(allAccounts -> {
-                new SelectedAccountResponsesCollector(mKeyringService, mJsonRpcService, coins,
-                        Arrays.asList(allAccounts.accounts))
-                        .getAccounts(mOrigin, defaultAccountPerCoin -> {
-                            defaultAccountPerCoins.call(defaultAccountPerCoin);
-                        });
-            });
-        }
-    }
-
-    void setOrigin(Origin origin) {
-        mOrigin = origin;
-    }
-
-    private void update(int coinType, String chainId) {
+    public void update() {
         synchronized (mLock) {
             if (mKeyringService == null) {
                 return;
             }
-            mKeyringService.getKeyringInfo(getSelectedCoinKeyringId(coinType, chainId),
-                    keyringInfo -> { _mSelectedCoinKeyringInfoLiveData.postValue(keyringInfo); });
             mKeyringService.getAllAccounts(allAccounts -> {
-                final List<AccountInfo> accountInfos = Arrays.asList(allAccounts.accounts);
-                _mAccountInfos.postValue(accountInfos);
-                if (coinType == CoinType.FIL) {
-                    mKeyringService.getFilecoinSelectedAccount(chainId, accountAddress -> {
-                        updateSelectedAccountAndState(accountAddress, accountInfos);
-                    });
-                } else {
-                    mKeyringService.getSelectedAccount(coinType, accountAddress -> {
-                        updateSelectedAccountAndState(accountAddress, accountInfos);
-                    });
-                }
+                _mAllAccountsInfo.postValue(allAccounts);
+                _mAccountInfos.postValue(Arrays.asList(allAccounts.accounts));
+                _mSelectedAccount.postValue(allAccounts.selectedAccount);
```

Comment:
This is basically all that is needed from KeyringModel: broadcasting  AllAccountsInfo and parts of it as LiveData. And also setSelectedAccount